src/main/java/org/baeldung/spring/CaptchaConfig.java

```diff
@@ -0,0 +1,22 @@
+package org.baeldung.spring;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.client.ClientHttpRequestFactory;
+import org.springframework.http.client.SimpleClientHttpRequestFactory;
+
+@Configuration
+@ComponentScan(basePackages = { "org.baeldung.captcha" })
+@ConditionalOnProperty(prefix = "google.recaptcha.key", name = { "site", "secret" })
+public class CaptchaConfig {
+    @Bean
+    public ClientHttpRequestFactory clientHttpRequestFactory() {
```

Comment:
Can use just use RestTemplate? Any benefits of using ClientHttpRequestFactory? I saw that you have this code under **Protecting Server Resources** section of the article. Not sure how are we protecting server resource? Is it the connect timeout and read timeout?